src/Makefile.am

```diff
@@ -9,15 +9,28 @@ CXXFLAGS = -Wall -g -O3 -std=c++11
 LDFLAGS = -lpthread
 
 if DIST_WS
-HUPCPP_FLAGS_1 = -DDIST_WS
+HC_FLAGS_1 = -DDIST_WS
```

Comment:
I would disagree. Right below:

if HUPCPP
HC_FLAGS_2 = -DHUPCPP -DHC_COMM_WORKER
else
HC_FLAGS_2 =
endif

It is very clear from the conditional of that if statement that these flags relate to HUPCPP. Even more importantly, they relate to HUPCPP but are applied to HC. The conditional tells you what the flag relates to or is controlled by, the variable name tells you what they apply to or are used to build. Just like below we have:

if X86
if OSX
libhcpp_la_SOURCES += \
fcontext/jump_i386_sysv_macho_gas.S \
fcontext/make_i386_sysv_macho_gas.S
else

It's clear that the thing controlling the variable being set is the platform we're running on (x86, OSX), but that the variable being set applies to some source code for "libhcpp". If instead you did:

x86_la_SOURCES += \
fcontext/jump_i386_sysv_macho_gas.S \
fcontext/make_i386_sysv_macho_gas.S

you actually end up conveying less information to the reader.